sharkfuser/include/fusilli/node/conv_node.h

```diff
@@ -203,6 +203,144 @@ class ConvFPropNode : public NodeCRTP<ConvFPropNode> {
   }
 };
 
+class ConvWGradNode : public NodeCRTP<ConvWGradNode> {
+public:
+  ConvWGradAttr convWGradAttr;
+
+  ConvWGradNode(ConvWGradAttr &&attr, const Context &ctx)
+      : NodeCRTP(ctx), convWGradAttr(std::move(attr)) {}
+
+  // MLIR assembly emitter helper methods.
+  std::string emitNodePreAsm() const override final;
+  std::string getOperandNamesAsm() const;
+  std::string getOperandTypesAsm() const;
+  std::string getResultNamesAsm() const;
+  std::string getResultTypesAsm() const;
+  std::string getStrideOpsAsm() const;
+  std::string getPaddingOpsAsm() const;
+  std::string getDilationOpsAsm() const;
+  std::string getPermuteDYOpsAsm() const;
+  std::string getPermuteXOpsAsm() const;
+  std::string getPermuteDWOpsAsm() const;
+  std::string getPermutedEmptyDWOpsAsm() const;
+
+  const std::string &getName() const override final {
+    return convWGradAttr.getName();
+  }
+  Type getType() const override final { return Type::WGrad; }
+
+  ErrorObject preValidateNode() const override final {
```

Comment:
Could you do a pass over the conv_fprop preValidateNode and see if any checks are worth having here? For example we should check the padding/stride/dilation ranks match that of the conv.